js/src/data/utils.test.js

```diff
@@ -65,4 +73,180 @@ describe( 'getReportQuery', () => {
 			).toHaveProperty( 'orderby', freeFields[ 0 ] );
 		} );
 	} );
+
+	it( 'When the query does not contain `programs` or `products`, should have no `ids` field', () => {
+		forAnyCategoryAndDateReference( ( category, dateReference ) => {
+			const query = {};
+
+			expect(
+				getReportQuery( category, 'free', query, dateReference ).ids
+			).toBeUndefined();
```

Comment:
:nail_care: If we are specing "should have no `ids` field", I'd rather use `not.toHaveProperty`
```suggestion
			expect(
				getReportQuery( category, 'free', query, dateReference )
			).not.toHaveProperty( 'id' );
```

Reply:
Same applies for the ones below, `like `.toHaveProperty( 'ids', query.programs )` - it gives more meaningful errors when fail

Reply:
> If we are specing "should have no `ids` field", I'd rather use `not.toHaveProperty`

Agreed, that's the correct way to verify this case.

> it gives more meaningful errors when fail

Thanks! Updated by https://github.com/woocommerce/google-listings-and-ads/pull/856/commits/9e3533f5a3b3c9f82e7ba7e67a9ad9ac1f7fa86a.